sdks/go/pkg/beam/core/monitoring/monitoring.go

```diff
@@ -341,3 +343,189 @@ func int64Distribution(count, sum, min, max int64) ([]byte, error) {
+			}
+			distributions[key] = value
+		} else if IsGauge(minfo) {
+			value, err := extractGaugeValue(r)
+			if err != nil {
+				log.Println(err)
+				continue
+			}
+			gauges[key] = value
+		} else {
+			log.Println("unknown metric type")
+		}
+	}
+	return counters, distributions, gauges
+}
+
+func mergeCounters(
+	attempted map[metrics.MetricKey]int64,
+	committed map[metrics.MetricKey]int64) []metrics.CounterResult {
+	res := make([]metrics.CounterResult, 0)
+
+	for k := range attempted {
+		v, ok := committed[k]
+		if !ok {
+			v = -1
+		}
+		res = append(res, metrics.CounterResult{Attempted: attempted[k], Committed: v, Key: k})
+	}
+	return res
+}
+
+func mergeDistributions(
+	attempted map[metrics.MetricKey]metrics.DistributionValue,
+	committed map[metrics.MetricKey]metrics.DistributionValue) []metrics.DistributionResult {
+	res := make([]metrics.DistributionResult, 0)
+
+	for k := range attempted {
+		v, ok := committed[k]
+		if !ok {
+			v = metrics.DistributionValue{}
+		}
+		res = append(res, metrics.DistributionResult{Attempted: attempted[k], Committed: v, Key: k})
+	}
+	return res
+}
+
+func mergeGauges(
+	attempted map[metrics.MetricKey]metrics.GaugeValue,
+	committed map[metrics.MetricKey]metrics.GaugeValue) []metrics.GaugeResult {
+	res := make([]metrics.GaugeResult, 0)
+
+	for k := range attempted {
+		v, ok := committed[k]
+		if !ok {
+			v = metrics.GaugeValue{}
+		}
+		res = append(res, metrics.GaugeResult{Attempted: attempted[k], Committed: v, Key: k})
+	}
+	return res
+}
+
+func extractKey(mi *pipepb.MonitoringInfo) (metrics.MetricKey, error) {
+	labels := newLabels(mi.GetLabels())
+	stepName := getStepName(labels)
+	if stepName == "" {
+		return metrics.MetricKey{}, fmt.Errorf("Failed to deduce Step from MonitoringInfo: %v", mi)
+	}
+	return metrics.MetricKey{Step: stepName, Name: labels.Name(), Namespace: labels.Namespace()}, nil
+}
+
+func extractCounterValue(reader *bytes.Reader) (int64, error) {
+	value, err := coder.DecodeVarInt(reader)
+	if err != nil {
+		return -1, err
+	}
+	return value, nil
+}
+
+func extractDistributionValue(reader *bytes.Reader) (metrics.DistributionValue, error) {
+	values, err := decodeMany(reader, 4)
+	if err != nil {
+		return metrics.DistributionValue{}, err
+	}
+	return metrics.DistributionValue{Count: values[0], Sum: values[1], Min: values[2], Max: values[3]}, nil
+}
+
+func extractGaugeValue(reader *bytes.Reader) (metrics.GaugeValue, error) {
+	values, err := decodeMany(reader, 2)
+	if err != nil {
+		return metrics.GaugeValue{}, err
+	}
+	return metrics.GaugeValue{Timestamp: time.Unix(0, values[0]*int64(time.Millisecond)), Value: values[1]}, nil
+}
+
+func newLabels(miLabels map[string]string) *metrics.Labels {
+	labels := metrics.UserLabels(
+		miLabels["PTRANSFORM"],
+		miLabels[pipepb.MonitoringInfo_NAMESPACE.String()],
+		miLabels[pipepb.MonitoringInfo_NAME.String()])
+	return &labels
+}
+
+func getStepName(labels *metrics.Labels) string {
+	return labels.Transform()
+}
```

Comment:
This function doesn't do anything other than serve as a comment. Consider simply calling labels.Transform() directly at the only call site.